src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -208,18 +209,19 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
 
                     //
                     // Avoid instance state modification
-                    Dictionary<string, ConfigurationSetting> applicationSettings = null;
                     Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;
-                    Dictionary<KeyValueIdentifier, KeyValueChange> keyValueChanges = null;
+                    List<KeyValueChange> keyValueChanges = null;
                     List<KeyValueChange> changedKeyValuesCollection = null;
+                    Dictionary<string, ConfigurationSetting> data = null;
+                    Dictionary<string, ConfigurationSetting> mappedData = null;
                     bool refreshAll = false;
                     StringBuilder logInfoBuilder = new StringBuilder();
                     StringBuilder logDebugBuilder = new StringBuilder();
 
                     await ExecuteWithFailOverPolicyAsync(availableClients, async (client) =>
                         {
-                            applicationSettings = null;
-                            keyValueChanges = new Dictionary<KeyValueIdentifier, KeyValueChange>();
+                            mappedData = null;

```

Comment:
Is it possible that you meant to nullify `data` here?

Reply:
Yea I think I missed changing that back to what it was before, should be data. I can also declare mappedData later since it's not used until after ExecuteWithFailOverPolicyAsync

Reply:
That's a good point. We probably dont even need the temp dictionary for mappedData because its always set outside ExecuteWithFailOverPolicyAsync. It looks like we could directly modify _mappedData at that point.

Reply:
Oh I see, if mappedData was altered then we would always set _mappedData equal to it later anyway.